### PR TITLE
Handle disconnects

### PR DESCRIPTION
**Pull request by pdcook**
Requires UnboundLib v2.10.0+: https://github.com/Rounds-Modding/UnboundLib/pull/59

### CODE REVIEW

**Comment by olavim** [comment]
Is this being merged to the correct branch?

**Comment by pdcook** [comment]
> Is this being merged to the correct branch?

Nope - good catch

**Review by olavim — COMMENTED** [review]
I see a lot of duplicated code between the two gamemode handlers. Also, since UnboundLib already reassigns teams, can't you just use them directly after letting `GameModeHandler.OnPlayerLeftRoom` process first (by calling `base.OnPlayerLeftRoom` before anything else) ?

**Comment by pdcook** [comment]
> I see a lot of duplicated code between the two gamemode handlers. Also, since UnboundLib already reassigns teams, can't you just use them directly after letting `GameModeHandler.OnPlayerLeftRoom` process first (by calling `base.OnPlayerLeftRoom` before anything else) ?

In regards to copying code: yes, I agree. The two gamemodes are actually identical, only their handlers differ. I'm thinking of making a `RWFGameMode` class that they can both inherit from, and that other gamemode mods that reference RWF could inherit from as well.

In regards to handling team reassignment in Unbound first - we can't do that since we need to know the old teamIDs/playerIDs at the same time as the new ones to properly fix `teamPoints` and `teamRounds`.

File: RoundsWithFriends/GameModes/DeathmatchHandler.cs [hunk 1]
```diff
@@ -71,28 +57,24 @@ public override void OnPlayerLeftRoom(Photon.Realtime.Player otherPlayer)
             {
                 if (!newTeamPoints.Keys.Contains(newTeamIDs[player]))
                 {
-                    newTeamPoints[newTeamIDs[player]] = GM_Deathmatch.instance.teamPoints[player.teamID];
+                    newTeamPoints[newTeamIDs[player]] = GM_TeamDeathmatch.instance.teamPoints[oldTeamIDs[player]];
```
Comment (olavim) [inline RoundsWithFriends/GameModes/DeathmatchHandler.cs]: Copy pasted the wrong gamemode here I think?

Comment (pdcook) [reply RoundsWithFriends/GameModes/DeathmatchHandler.cs]: good catch again - though this does become irrelevant in the game mode abstraction PR. but I'll fix it here too

**Review by olavim — APPROVED** [review]
looks good

**Pull request merged**